### PR TITLE
fixed isValidLogin when Backendless.Async

### PR DESCRIPTION
**Pull request by Valodya**
fixed a problem when we pass Backendless.Async into isValidLogin method of UserService

### CODE REVIEW

File: libs/backendless.js [hunk 1]
```diff
@@ -1923,7 +1923,18 @@
...
-                return !!user;
+                if (async) {
+                    //if async need to put it to the end of the stack
+                    setTimeout(function() {
+                        if (!!user) {
+                            responder.success(user);
+                        } else {
+                            responder.fault();
+                        }
+                    }, 1);
```
Comment (vengrov) [inline libs/backendless.js]: `0` is fine I suppose

File: libs/backendless.js [hunk 2]
```diff
@@ -1923,7 +1923,18 @@
             } else {
                 var user = Backendless.UserService.getCurrentUser();
 
-                return !!user;
+                if (async) {
+                    //if async need to put it to the end of the stack
+                    setTimeout(function() {
+                        if (!!user) {
+                            responder.success(user);
```
Comment (vengrov) [inline libs/backendless.js]: Did you mean `!!user` ?

File: libs/backendless.js (hunk 1 above)
Comment (Valodya) [inline libs/backendless.js]: yes, you are right

File: libs/backendless.js (hunk 2 above)
Comment (Valodya) [inline libs/backendless.js]: I think that success method will not be called, because if we have no `user-token` that's mean the user is not valid, right?

Comment (vengrov) [inline libs/backendless.js]:
```
setTimeout(function() {
  responder.success(!!user);
}, 0);
```

**Pull request merged**